skyportal/models/analysis.py

```diff
@@ -553,3 +657,75 @@ def delete_assoc_analysis_data_from_disk(mapper, connection, target):
     for analysis in target.obj_analyses:
         log(f' ... deleting analysis data for analysis id={analysis.id}')
         analysis.delete_data()
+
+
+@event.listens_for(Classification, 'after_insert')
+def create_default_analysis(mapper, connection, target):
+    log(f'Checking for default analyses for classification {target.id}')
+
+    @event.listens_for(inspect(target).session, "after_flush", once=True)
+    def receive_after_flush(session, context):
+        try:
+            from skyportal.handlers.api.analysis import post_analysis
+
+            target_data = target.to_dict()
+
+            stmt = sa.select(DefaultAnalysis).where(
+                DefaultAnalysis.source_filter['classifications'].contains(
+                    [{"name": target_data['classification']}]
+                )
+            )
+            default_analyses = session.scalars(stmt).all()
+
+            for default_analysis in default_analyses:
+                classification_filter = next(
+                    (
+                        classification
+                        for classification in default_analysis.source_filter[
+                            "classifications"
+                        ]
+                        if classification["name"] == target_data['classification']
```

Comment:
we may want to do a "like" instead of a "equals" here. I can imagine a case where someone would want to trigger an analysis if the classification matches "supernova".